internal/client.go

```diff
@@ -326,7 +327,7 @@ func NewClient(service workflowserviceclient.Interface, domain string, options *
 	return &workflowClient{
 		workflowService: metrics.NewWorkflowServiceWrapper(service, metricScope),
 		domain:          domain,
-		metricsScope:    metricScope,
+		metricsScope:    &metrics.TaggedScope{Scope: metricScope, Map: &sync.Map{}},
```

Comment:
create a func to create this because who using this taggedScope should not need to know that it needs a sync.Map internally:
func newTaggedScope(scope tally.Scope) *TaggedScope {
  return &metrics.TaggedScope{Scope: scope, Map: &sync.Map{}}
}